app/src/androidTest/java/com/duckduckgo/app/feedback/ui/common/FeedbackViewModelTest.kt

```diff
@@ -0,0 +1,315 @@
+
+    @Before
+    @UiThreadTest
+    fun setup() {
+        testee = FeedbackViewModel(playStoreUtils, feedbackSubmitter)
+        testee.command.observeForever(commandObserver)
+    }
+
+    @After
+    @UiThreadTest
+    fun tearDown() {
+        testee.command.removeObserver(commandObserver)
+    }
+
+    @UiThreadTest
+    @Test
+    fun whenInitialisedThenFragmentStateIsForFirstStep() {
+        assertTrue(fragmentViewState is InitialAppEnjoymentClarifier)
+    }
+
+    @Test
+    @UiThreadTest
+    fun whenCanRateAppAndUserSelectsInitialHappyFaceThenFragmentStateIsFirstStepOfHappyFlow() {
+        configureRatingCanBeGiven()
+        testee.userSelectedPositiveFeedback()
+        assertTrue(fragmentViewState is FragmentState.PositiveFeedbackFirstStep)
+        verifyForwardsNavigation(fragmentViewState)
+    }
+
+    @Test
+    @UiThreadTest
+    fun whenCannotRateAppAndUserSelectsInitialHappyFaceThenFragmentStateSkipsStraightToSharingFeedback() {
+        configureRatingCannotBeGiven()
+        testee.userSelectedPositiveFeedback()
+        assertTrue(fragmentViewState is FragmentState.PositiveShareFeedback)
+        verifyForwardsNavigation(fragmentViewState)
+    }
+
+    @Test
+    @UiThreadTest
+    fun whenCanRateAppAndUserNavigatesBackFromPositiveInitialFragmentThenFragmentStateIsInitialFragment() {
+        configureRatingCanBeGiven()
+        testee.userSelectedPositiveFeedback()
+        testee.onBackPressed()
+
+        assertTrue(fragmentViewState is InitialAppEnjoymentClarifier)
+        verifyBackwardsNavigation(fragmentViewState)
+    }
+
+    @Test
+    @UiThreadTest
+    fun whenCannotRateAppAndUserNavigatesBackFromPositiveInitialFragmentThenFragmentStateIsInitialFragment() {
+        configureRatingCannotBeGiven()
+        testee.userSelectedPositiveFeedback()
+        testee.onBackPressed()
+
+        assertTrue(fragmentViewState is InitialAppEnjoymentClarifier)
+    }
+
+
+    @Test
+    fun whenUserChoosesNotToProvideFurtherDetailsForPositiveFeedbackThenSubmitted() = runBlocking<Unit> {
+        withContext(Dispatchers.Main) {
+            testee.userGavePositiveFeedbackNoDetails()
+        }
+
+        verify(feedbackSubmitter).sendPositiveFeedback(null)
+    }
+
+    @Test
+    fun whenUserChoosesNotToProvideFurtherDetailsForPositiveFeedbackThenExitCommandIssued() = runBlocking<Unit> {
+        withContext(Dispatchers.Main) {
+            testee.userGavePositiveFeedbackNoDetails()
+        }
+
+        val command = captureCommand() as Command.Exit
+        assertTrue(command.feedbackSubmitted)
+    }
+
+    @Test
+    fun whenUserProvidesFurtherDetailsForPositiveFeedbackThenFeedbackSubmitted() = runBlocking<Unit> {
+        withContext(Dispatchers.Main) {
+            testee.userProvidedPositiveOpenEndedFeedback("foo")
+        }
+
+        verify(feedbackSubmitter).sendPositiveFeedback("foo")
+    }
+
+    @Test
+    fun whenUserProvidesFurtherDetailsForPositiveFeedbackThenExitCommandIssued() = runBlocking<Unit> {
```

Comment:
This test is intermittently failing, see https://app.bitrise.io/build/68a947da5eaa70d9.

Reply:
Made a change; I've ran the CI quite a few times in the hopes that it's now stable.